src/Orleans.Core/Runtime/OutsideRuntimeClient.cs

```diff
@@ -335,7 +335,11 @@ public void ReceiveResponse(Message response)
 
         private void UnregisterCallback(CorrelationId id)
         {
-            callbacks.TryRemove(id, out _);
+            if (!callbacks.TryRemove(id, out _))
```

Comment:
This is not an error - a callback might timeout and receive a response concurrently. It is a natural race